src/main/java/com/develhope/spring/model/user/Rental.java

```diff
@@ -0,0 +1,47 @@
+package com.develhope.spring.model.user;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+
+public class Rental {
+    private LocalDate startDate;
+    private LocalDate endDate;
+    private BigDecimal dailyCost;
+    private BigDecimal totalCost;
+    private boolean paid;
+
+    //it was decided to use vehicle (object) but instead of using
+    // Vehicle as object is created private int vehicleId;
+    // passed to the constructor
+    // handled the same way in the Order class of User
+    private int vehicleId;
+
+
+    public Rental(LocalDate startDate, LocalDate endDate, BigDecimal dailyCost, int vehicleId) {
+        this.startDate = startDate;
+        this.endDate = endDate;
+        this.dailyCost = dailyCost;
+        this.vehicleId = vehicleId;
+        this.calculateTotalCost();
```

Comment:
non serve il this in questo caso, puoi lasciare direttamente calculateTotalCost() a questa riga. Il this.totalCost sta già all'interno del metodo.

Reply:
se non compromette il funzionamento ( e non credo visto che punta in ogni caso al metodo della classe non lo cambio.

Reply:
Capisco ma appunto visto che non cambia rimane più pulito senza.

Reply:
eliminato, effettivamente il codice è più pulito anche se il this era esplicativo.